lightning/src/ln/peers/handshake/acts.rs

```diff
@@ -0,0 +1,33 @@
+pub struct ActOne(
+	pub(super) [u8; 50]
+);
+
+pub struct ActTwo(
+	pub(super) [u8; 50]
+);
+
+pub struct ActThree(
+	pub(super) [u8; 66]
+);
+
+pub enum Act {
+	One(ActOne),
+	Two(ActTwo),
+	Three(ActThree),
+}
+
+impl Act {
+	pub fn serialize(&self) -> Vec<u8> {
```

Comment:
Can we name this something else? I almost responded with "why the hell would you ever want to serialize an Act for storage?" before I realized it meant "write act to a vec to send to a peer" (also, can we not return a slice here?).

Reply:
Perhaps `to_vec`? We could return a slice, but would have to mark its lifetime as linked to the act object's.

Reply:
What's wrong with using lifetimes? The callsite can chose to store it in a Vec or copy to some local buffer if they wont want it.